crates/consensus-logic/src/checkpoint_verification.rs

```diff
@@ -82,50 +82,28 @@ pub fn verify_proof(
     proof_receipt: &ProofReceipt,
     rollup_params: &RollupParams,
 ) -> ZkVmResult<()> {
-    let rollup_vk = rollup_params.rollup_vk;
     let checkpoint_idx = checkpoint.batch_info().epoch();
     info!(%checkpoint_idx, "verifying proof");
 
-    // FIXME: we are accepting empty proofs for now (devnet) to reduce dependency on the prover
-    // infra.
-    let allow_empty = rollup_params.proof_publish_mode.allow_empty();
-    let is_empty_proof = proof_receipt.proof().is_empty();
-    let accept_empty_proof = is_empty_proof && allow_empty;
-    let skip_public_param_check = proof_receipt.public_values().is_empty() && allow_empty;
-    let is_non_native_vk = !matches!(rollup_vk, RollupVerifyingKey::NativeVerifyingKey(_));
-
-    if !skip_public_param_check {
-        let expected_public_output = *checkpoint.batch_transition();
-        let actual_public_output: BatchTransition =
-            borsh::from_slice(proof_receipt.public_values().as_bytes())
-                .map_err(|e| ZkVmError::OutputExtractionError { source: e.into() })?;
-
-        if expected_public_output != actual_public_output {
-            dbg!(actual_public_output, expected_public_output);
-            return Err(ZkVmError::ProofVerificationError(
-                "Public output mismatch during proof verification".to_string(),
-            ));
-        }
-    }
+    // Do the public parameters check
+    let expected_public_output = *checkpoint.batch_transition();
+    let actual_public_output: BatchTransition =
+        borsh::from_slice(proof_receipt.public_values().as_bytes())
+            .map_err(|e| ZkVmError::OutputExtractionError { source: e.into() })?;
 
-    if accept_empty_proof && is_non_native_vk {
-        warn!(%checkpoint_idx, "verifying empty proof as correct");
-        return Ok(());
-    }
-
-    if !allow_empty && is_empty_proof {
-        return Err(ZkVmError::ProofVerificationError(format!(
-            "Empty proof received for checkpoint {checkpoint_idx}, which is not allowed in strict proof mode. \
-            Check `proof_publish_mode` in rollup_params; set it to a non-strict mode (e.g., `timeout`) to accept empty proofs."
-        )));
+    if expected_public_output != actual_public_output {
+        dbg!(actual_public_output, expected_public_output);
+        return Err(ZkVmError::ProofVerificationError(
+            "Public output mismatch during proof verification".to_string(),
+        ));
```

Comment:
Should make a ticket to eliminate this duplication in our serialized proof structure.  The proof that we put on-chain should not have this receipt structure, we would construct this receipt from the fields in the checkpoint which we already just compare the receipt to.

Since we're changing the proof structure to support the ASM-based flow, we can probably wait to do that and maybe do it as part of the integration work there.